src/components/AvatarStack/Avatars.tsx

```diff
@@ -1,66 +1,95 @@
+            key={user.clientId}
+            style={{
+              right: rightOffset,
+              zIndex: users.length - index,
+            }}
+          >
             <div
-              className="absolute right-0 flex flex-col items-center"
-              key={user.clientId}
-              style={{
-                right: rightOffset,
-                zIndex: otherUsers.length - index,
-              }}
+              className={`${
+                user.isConnected ? user.profileData.memberColor : "bg-gray-200"
+              } h-12 w-12 rounded-full flex items-center justify-center relative border-2 border-gray-200`}
+              onMouseOver={() => setHoveredClientId(user.clientId)}
+              onMouseLeave={() => setHoveredClientId(null)}
+              id="avatar"
             >
-              <UserCircleIcon className="absolute mt-2 h-8 w-8 opacity-80 stroke-white fill-transparent pointer-events-none" />
+              <p
+                className={`relative z-20 text-xs ${
+                  user.isConnected ? "text-white" : "text-gray-400"
+                }`}
+              >
+                {user.profileData.name
+                  .split(" ")
+                  .map((word: string) => word.charAt(0))
+                  .join("")}
+              </p>
               <div
-                className={`bg-gradient-to-l ${colours[index]} h-12 w-12 rounded-full mb-2 shadow-[0_0_0_4px_rgba(255,255,255,1)]`}
-                onMouseOver={() => setHoveredClientId(user.clientId)}
-                onMouseLeave={() => setHoveredClientId(null)}
-              ></div>
-              {!user.isConnected ? (
-                <div className="absolute top-0 h-12 w-12 rounded-full mb-2 bg-white opacity-80 pointer-events-none" />
-              ) : null}
-              {hoveredClientId === user.clientId ? (
-                <div className="absolute top-14 min-w-[175px] px-4 py-2 bg-black rounded-lg text-white">
-                  <UserInfo user={user} />
-                </div>
-              ) : null}
+                className={`${
+                  user.isConnected ? "bg-green-500" : "bg-gray-400"
+                } w-[10px] h-[10px] rounded-full absolute bottom-1 left-0 transform translate-y-1/2 translate-x-1/2 z-10`}
+                id="status-indicator"
+              />
             </div>
-          );
-        })}
+
+            {hoveredClientId === user.clientId ? (
+              <div className="absolute -top-16 min-w-[190px] px-2 py-2 bg-black rounded-lg text-white">
+                <UserInfo user={user} />
+              </div>
+            ) : null}
+          </div>
+        );
+      })}
     </>
   );
 };
+const Avatars = ({ otherUsers }: { otherUsers: SpaceMember[] }) => {
+  const OVERLAP_AMOUNT = 40;
+  const AVATAR_WIDTH = 48;
+  const totalWidth = AVATAR_WIDTH + OVERLAP_AMOUNT * otherUsers.length;
+
+  return (
+    <div
+      className="relative flex max-w-[206px] "
+      style={{ width: `${totalWidth}px` }}
```

Comment:
do we actually need `px` here, I think we can just do the number and it will do the px on itself

Reply:
afaik, this is needed unless you're doing it in plain HTML/CSS. With React & Vanilla JS you do need to explicitly state it.